File: my-folder/0088-merge-sorted-array/solution.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        # t=n
        
        # if len(nums1)==1 and len(nums2)==1:
        #     nums1[0]=max(nums1[0],nums2[0])
            
        # for i in range(n):
        #     nums1[i]=nums1[i]
        #     nums1[i+1]=nums1[t]
        #     t+=1
        # print(modified_nums1)
        for i in range(len(nums2)):
            nums1[i+m]=nums2[i]
        nums1.sort()

File: my-folder/0088-merge-sorted-array/solution.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int left = m - 1;
        int right = n - 1;
        // idx to keep track of the position until where we sorted already
        int idx = m + n - 1;

        while(left >= 0 && right >= 0){
            if(nums1[left] > nums2[right]){
                nums1[idx--] = nums1[left--];
            }
            else{
                nums1[idx--] = nums2[right--];
            }
        }

        while(right >= 0){
            nums1[idx--] = nums2[right--];
        }
    }
}
